### PR TITLE
Update MSRV to 1.48. Fixes CI

### PR DESCRIPTION
**Pull request by sanket1729**

### CODE REVIEW

**Comment by apoelstra** [comment]
Can you update the fuzzer rustc to 1.64 from 1.58, and also update `generate-files.sh` to reflect the `regex = "=1.4.0"` thing (if that is still necessary)?

**Comment by sanket1729** [comment]
Nice. Found a new fuzz failure

**Review by tcharding — COMMENTED** [review]
LGTM, glad to see you worked out the pinning, I tried and failed yesterday. (FWIW I used LGTM instead of ACK because I'm assuming you are going to squash the `fix!` commits.)

**Comment by sanket1729** [comment]
After a lot of iteration on fuzzer, this is finally ready for review.

File: .gitconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[remote "upstream"]
+        fetch = +refs/pull/*:refs/remotes/pr/*
+        fetch = +refs/merge-requests/*:refs/remotes/pr/*
```
Comment (apoelstra) [inline .gitconfig]: In 3ecd2fbccc64bc707a9f418c93e019192e9113d9:

You should not have committed this file

Comment (apoelstra) [reply .gitconfig]: it's deleted in the next commit, but could still cause weird things to happen during bisection.

File: src/policy/compiler.rs [hunk 2]
```diff
@@ -1048,6 +1056,7 @@ where
 /// Helper function to compile different types of binary fragments.
 /// `sat_prob` and `dissat_prob` represent the sat and dissat probabilities of
 /// root or. `weights` represent the odds for taking each sub branch
+#[allow(clippy::too_many_arguments)]
```
Comment (apoelstra) [inline src/policy/compiler.rs:1059]: In e491308b6075d5274917750ad7258cb604b99aea:

I think this is fine, but FYI both this and the "too many arguments" lint thresholds can be set globally in clippy.toml.

**Review by apoelstra — DISMISSED** [review]
ACK 13950fd16161afd9b28345c68192aff997048231

**Comment by apoelstra** [comment]
Oh, actually, I'd like you to remove the spurious file before merging.

File: .gitconfig (hunk 1 above)
Comment (sanket1729) [reply .gitconfig]: Removed

**Comment by sanket1729** [comment]
Added another commit that adds a static vector

**Review by apoelstra — APPROVED** [review]
ACK 60fde9e9519125d3b1ecccea302a0a6e6ce484fa

**Review by apoelstra — APPROVED** [review]
ACK 6b76997b14fd606b6ceba884012512c06200a4a4

**Pull request merged**